src/pocket/PocketApiGatewayLambdaIntegration.ts

```diff
@@ -0,0 +1,106 @@
+}
+
+interface InitializedGatewayRoute {
+  resource: APIGateway.ApiGatewayResource;
+  method: APIGateway.ApiGatewayMethod;
+  integration: APIGateway.ApiGatewayIntegration;
+  permission: LambdaFunction.LambdaPermission;
+}
+
+export class PocketApiGateway extends Resource {
+  private readonly config: PocketApiGatewayProps;
+  private apiGatewayRestApi: APIGateway.ApiGatewayRestApi;
+  private routes: InitializedGatewayRoute[];
+  private apiGatewayDeployment: APIGateway.ApiGatewayDeployment;
+  private apiGatewayStage: APIGateway.ApiGatewayStage;
+
+  constructor(scope: Construct, name: string, config: PocketApiGatewayProps) {
+    super(scope, name);
+    this.apiGatewayRestApi = new APIGateway.ApiGatewayRestApi(
+      scope,
+      `api-gateway-rest`,
+      {
+        name: config.name,
+      }
+    );
+    // do the deployment after everything is initialized but before adding lambda permissions??
+    // https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_deployment
+    this.createLambdaIntegrations(config);
+  }
+  // loop over routes and generate methods
+  /**
+   * Generate aws proxy routes handled by aws lambda
+   * @param routes
+   */
+  private createLambdaIntegrations(config: PocketApiGatewayProps) {
+    this.routes = config.routes.map((route: ApiGatewayLambdaRoute) => {
+      const lambda = new PocketVersionedLambda(
+        this,
+        `${route.path}-lambda`,
+        route.eventHandler
+      );
+      const resource = new APIGateway.ApiGatewayResource(this, route.path, {
+        parentId: this.apiGatewayRestApi.rootResourceId,
+        pathPart: route.path,
+        restApiId: this.apiGatewayRestApi.id,
+      });
+      const method = new APIGateway.ApiGatewayMethod(
+        this,
+        `${route.path}-method`,
+        {
+          restApiId: this.apiGatewayRestApi.id,
+          resourceId: resource.id,
+          // authorization: route.authorizationType,
+          authorization: 'NONE',
+          httpMethod: route.method,
+        }
+      );
+      const integration = new APIGateway.ApiGatewayIntegration(
+        this,
+        `${route.path}-integration`,
+        {
+          httpMethod: route.method,
+          integrationHttpMethod: 'POST', // lambda has to be post
+          resourceId: resource.id,
+          restApiId: this.apiGatewayRestApi.id,
+          type: 'AWS_PROXY',
+          uri: lambda.lambda.versionedLambda.invokeArn,
+        }
+      );
+      const permission = new LambdaFunction.LambdaPermission(
+        this,
+        `${route.path}-allow-gateway-lambda-invoke`,
+        {
+          functionName: lambda.lambda.versionedLambda.functionName,
+          action: 'lambda:InvokeFunction',
+          principal: 'apigateway.amazonaws.com',
+          // The /*/*/* part allows invocation from any stage, method and resource path
+          // within API Gateway REST API.
+          sourceArn: `${this.apiGatewayRestApi.executionArn}/*/*/*`,
```

Comment:
just checking if the src arn for invokeFunction is api gateway arn or lambda arn
https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/AuroraMySQL.Integrating.Authorizing.IAM.LambdaCreatePolicy.html

Reply:
might need more context on api gateway, so in this case lambda resides within api gateway?

Reply:
The Lambda is a separate resource. This is a permission to invoke that lambda from the api gateway. The docs you linked are from aurora -- see [here](https://docs.aws.amazon.com/lambda/latest/dg/services-apigateway.html#apigateway-permissions) for more information.